app/controllers/documents_controller.rb

```diff
@@ -101,7 +101,16 @@ def document_error_messages
   end
 
   def fetch_document
-    @document = document_klass.find(params[:content_id])
+    begin
+      @document = document_klass.find(params[:content_id])
+    rescue Document::RecordNotFound => e
+      flash[:danger] = "Document not found"
+      redirect_to documents_path(document_type: document_type)
```

Comment:
Seems a bit strange to do a flash/redirect dance, any reason not to just issue a 404?  Or is it because this is an admin app, not a frontend one?